docs: clarify supabase_admin default privileges (#47952)
supabase/supabase: apps/docs/content/guides/api/securing-your-api.mdx

  revoke execute on functions from public;
```

<Admonition type="tip">

> The default privileges shown in [pg_default_acl](https://www.postgresql.org/docs/current/catalog-pg-default-acl.html) that originate from [supabase_admin](/docs/guides/database/postgres/roles#supabaseadmin) (granting to anon, authenticated, and service_role) are intentional and part of Supabase's standard permission model. The supabase_admin role is an internal management role that cannot authenticate through the Data API, so it poses no direct security risk.

Reviewer: ```suggestion
These default privileges pose no direct security risk. They appear in [pg_default_acl](https://www.postgresql.org/docs/current/catalog-pg-default-acl.html), granted by [supabase_admin](/docs/guides/database/postgres/roles#supabaseadmin) to `anon`, `authenticated`, and `service_role`. The default privileges are intentional and part of Supabase's standard permission model. The `supabase_admin` role is an internal management role that can't authenticate through the Data API.
```
This suggestion:
- Brings (what I assume to be) the 'so what' to the very front
- Simplifies sentence structure
- Removes parenthetical

You may need to check again to see that I haven't accidentally altered the meaning.

I agree that this belongs as a 'tip' or a 'note'. It belongs as an aside but is not urgent or a warning. Just addresses a concern and context.

I do wonder if there is any information redundant here or if this can be shortened even more. The mention to granting to "`anon`, `authenticated`, and `service_role`" is only a couple of paragraphs above.

Check with Claude if roles like 'anon' are normally in tick marks.
